The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5358247138/jobs/9720264360) [(plain)](https://github.com/rust-lang/rust/commit/75ac445ac3a255104a25d7c54522c3887d26719e/checks/14507434459/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [ui] tests/ui/traits/new-solver/alias-bound-unsound.rs stdout ----
diff of stderr:

19 LL |     drop(<() as Foo>::copy_me(&x));
21 
- error: aborting due to 2 previous errors
- error: aborting due to 2 previous errors
+ error: the type `_` is not well-formed
+    |
+    |
+ LL |     drop(<() as Foo>::copy_me(&x));
+ 
+ 
+ error: the type `_` is not well-formed
+    |
+    |
+ LL |     drop(<() as Foo>::copy_me(&x));
+ 
+ error: aborting due to 4 previous errors
23 
24 For more information about this error, try `rustc --explain E0277`.
---
To only update this specific test, also pass `--test-args traits/new-solver/alias-bound-unsound.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/traits/new-solver/alias-bound-unsound.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/traits/new-solver/alias-bound-unsound" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/traits/new-solver/alias-bound-unsound/auxiliary" "-Ztrait-solver=next"
stdout: none
--- stderr -------------------------------
error[E0277]: the trait bound `<() as Foo>::Item: Copy` is not satisfied
##[error]  --> fake-test-src-base/traits/new-solver/alias-bound-unsound.rs:23:10
   |
LL |     drop(<() as Foo>::copy_me(&x));
   |          ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `<() as Foo>::Item`
note: required by a bound in `Foo::Item`
  --> fake-test-src-base/traits/new-solver/alias-bound-unsound.rs:10:30
   |
LL |     type Item: Copy
LL |     type Item: Copy
   |          ---- required by a bound in this associated type
LL |     where
LL |         <Self as Foo>::Item: Copy;
   |                              ^^^^ required by this bound in `Foo::Item`

error: the type `<() as Foo>::Item` is not well-formed
##[error]  --> fake-test-src-base/traits/new-solver/alias-bound-unsound.rs:23:10
   |
LL |     drop(<() as Foo>::copy_me(&x));

error: the type `_` is not well-formed
##[error]  --> fake-test-src-base/traits/new-solver/alias-bound-unsound.rs:23:5
   |
   |
LL |     drop(<() as Foo>::copy_me(&x));

error: the type `_` is not well-formed
##[error]  --> fake-test-src-base/traits/new-solver/alias-bound-unsound.rs:23:10
   |
   |
LL |     drop(<() as Foo>::copy_me(&x));

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0277`.
```

</details>

--- a/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs
+++ b/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs
@@ -1049,6 +1049,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     }
 
                     ty::PredicateKind::Clause(ty::ClauseKind::WellFormed(ty)) => {
+                        let ty = self.resolve_vars_if_possible(ty);
                         match self.tcx.sess.opts.unstable_opts.trait_solver {
                             TraitSolver::Classic => {
                                 // WF predicates cannot themselves make

--- a/tests/ui/for/issue-20605.next.stderr
+++ b/tests/ui/for/issue-20605.next.stderr
@@ -14,7 +14,7 @@ LL |     for item in *things { *item = 0 }
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature
 
-error: the type `<_ as IntoIterator>::IntoIter` is not well-formed
+error: the type `<dyn Iterator<Item = &'a mut u8> as IntoIterator>::IntoIter` is not well-formed
   --> $DIR/issue-20605.rs:5:17
    |
 LL |     for item in *things { *item = 0 }

--- a/tests/ui/for/issue-20605.rs
+++ b/tests/ui/for/issue-20605.rs
@@ -4,7 +4,7 @@
 fn changer<'a>(mut things: Box<dyn Iterator<Item=&'a mut u8>>) {
     for item in *things { *item = 0 }
     //~^ ERROR the size for values of type
-    //[next]~^^ ERROR the type `<_ as IntoIterator>::IntoIter` is not well-formed
+    //[next]~^^ ERROR the type `<dyn Iterator<Item = &'a mut u8> as IntoIterator>::IntoIter` is not well-formed
     //[next]~| ERROR the trait bound `dyn Iterator<Item = &'a mut u8>: IntoIterator` is not satisfied
 }
 

--- a/tests/ui/traits/new-solver/alias-bound-unsound.rs
+++ b/tests/ui/traits/new-solver/alias-bound-unsound.rs
@@ -23,7 +23,7 @@ fn main() {
     drop(<() as Foo>::copy_me(&x));
     //~^ ERROR `<() as Foo>::Item: Copy` is not satisfied
     //~| ERROR `<() as Foo>::Item` is not well-formed
-    //~| ERROR `_` is not well-formed
-    //~| ERROR `_` is not well-formed
+    //~| ERROR `<() as Foo>::Item` is not well-formed
+    //~| ERROR `<() as Foo>::Item` is not well-formed
     println!("{x}");
 }

--- a/tests/ui/traits/new-solver/alias-bound-unsound.stderr
+++ b/tests/ui/traits/new-solver/alias-bound-unsound.stderr
@@ -19,13 +19,13 @@ error: the type `<() as Foo>::Item` is not well-formed
 LL |     drop(<() as Foo>::copy_me(&x));
    |          ^^^^^^^^^^^^^^^^^^^^^^^^
 
-error: the type `_` is not well-formed
+error: the type `<() as Foo>::Item` is not well-formed
   --> $DIR/alias-bound-unsound.rs:23:5
    |
 LL |     drop(<() as Foo>::copy_me(&x));
    |     ^^^^
 
-error: the type `_` is not well-formed
+error: the type `<() as Foo>::Item` is not well-formed
   --> $DIR/alias-bound-unsound.rs:23:10
    |
 LL |     drop(<() as Foo>::copy_me(&x));

